### PR TITLE
api code for PUT and created WorkshopDeleteView

### PR DESCRIPTION
**Pull request by mkeerthiasm**
Added API code for PUT- tested & it's working
Added API code for DELETE-haven't tested yet!

### CODE REVIEW

File: exception_al/workshops/views.py [hunk 1]
```diff
@@ -49,19 +53,41 @@ def get(self, request, pk):
...
     
+    def delete(self, request, pk):
+        workshop = self.get_object(pk)
+        workshop.delete()
+        return Response(status=status.HTTP_204_NO_CONTENT)
+    
+    
+
+class WorkshopDeleteView(APIView):
+    permission_classes = [permissions.IsAuthenticatedOrReadOnly]
```
Comment (suanTech) [inline exception_al/workshops/views.py:71]: Please remove this view, as the WorkshopDetailView already contains Delete method.

File: exception_al/workshops/urls.py [hunk 2]
```diff
@@ -1,12 +1,11 @@
...
-
-
+from .views import WorkshopListView, WorkshopDetailView, WorkshopViewSet, WorkshopDeleteView # This imports the WorkshopListView and WorkshopDetailView views from the views.py file in the workshops app
 urlpatterns = [
     # path('workshops/', WorkshopCreateView.as_view(), name='workshop-create'), # This maps the URL /workshops/ to the WorkshopCreateView view
     path('workshops/', WorkshopListView.as_view(), name='workshops'), # This maps the URL /workshops/ to the WorkshopListView view
     path('workshops/<int:pk>/', WorkshopDetailView.as_view(), name='workshop-detail'), # This maps the URL /workshops/<int:pk>/ to the WorkshopDetailView view
     path('workshops/<str:workshop_category>/', WorkshopViewSet.as_view({'get': 'list'}), name='workshop-list'), # This maps the URL /workshops/<str:workshop_category>/ to the WorkshopViewSet view
+    path('workshops/<int:pk>/delete/', WorkshopDeleteView.as_view(), name='workshop-delete'), # This maps the URL /workshops/<int:pk>/delete/ to the WorkshopDeleteView view
```
Comment (suanTech) [inline exception_al/workshops/urls.py:8]: WorkshopDetailView already has a delete method that takes ink:pk as a parameter, no need to add a separate delete view!

File: exception_al/workshops/views.py [hunk 3]
```diff
@@ -49,19 +53,41 @@ def get(self, request, pk):
 
     def put(self, request, pk):
         workshop = self.get_object(pk)
-        serializer = WorkshopSerializer(workshop, data=request.data)
+        serializer = WorkshopSerializer(data=request.data, context={'request': request})
+        # serializer = WorkshopSerializer(workshop, data=request.data)
```
Comment (suanTech) [inline exception_al/workshops/views.py:57]: Please remove comment

File: exception_al/workshops/views.py [hunk 4]
```diff
@@ -31,6 +32,9 @@ def post(self, request):
             return Response(serializer.data, status=status.HTTP_201_CREATED)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
     
+
+    
+
```
Comment (suanTech) [inline exception_al/workshops/views.py:37]: For below WorkshopDetailView, modify permissions to IsOwnerOrReadonly permission that you created. You'll need to import the custom permission from permissions.py

**Review by suanTech — CHANGES_REQUESTED** [review]
Good work Keerthi! A few things need attention :)

File: exception_al/workshops/views.py (hunk 1 above)
Comment (mkeerthiasm) [reply exception_al/workshops/views.py:71]: Done removed WorkshopDeleteView

File: exception_al/workshops/urls.py (hunk 2 above)
Comment (mkeerthiasm) [reply exception_al/workshops/urls.py:8]: removed the delete method

File: exception_al/workshops/views.py (hunk 3 above)
Comment (mkeerthiasm) [reply exception_al/workshops/views.py:57]: Done!

File: exception_al/workshops/views.py (hunk 4 above)
Comment (mkeerthiasm) [reply exception_al/workshops/views.py:37]: Done!!
Imported custom permissions from permissions.py
modified permissions to IsAuthenticatedOr ReadOnly, CustomPermission

**Pull request merged**